commands/util/pvp-rank.js

```diff
@@ -0,0 +1,161 @@
+    `&min_iv=${ivFilter}` +
+    `&att_iv=${this.inputATKiv}` +
+    `&def_iv=${this.inputDEFiv}` +
+    `&sta_iv=${this.inputSTAiv}`;
+
+    let $ = await fetch(this.url)
+      .then(res => res.text())
+      .then(body => cheerio.load(body))
+      .catch(err => log.error(err));
+    let ivScrapeData = [];
+    let siteData;
+    $("tr[class^=\"table-\"]").children().each(function(){
+      siteData = $(this).text();
+      ivScrapeData.push(siteData);
+      //console.log(typeof(siteData), siteData)
+    })
+    console.log(typeof(this.inputPokemon), this.inputPokemon.pbName),
+    this.name = this.inputPokemon.titleCase();
+    this.rank = ivScrapeData[0];
+    this.level = ivScrapeData[1];
+    this.cp = ivScrapeData[3];
+    this.atk = ivScrapeData[4];
+    this.def = ivScrapeData[5];
+    this.sta = ivScrapeData[6];
+    this.statproduct = ivScrapeData[8];
+  }
+
+  async displayInfo(message,args,command){
+    String.prototype.titleCase = function () {
+      return this.replace(/\w\S*/g, function(str) {return str.charAt(0).toUpperCase() + str.substr(1).toLowerCase();});
+    };
+
+    let league = '';
+    if(command == 'rank'){
+      league = 'GREAT';
+    }
+    else{
+      league = command.toUpperCase();
+    }
+    let requestInfo = `\n**${league} LEAGUE\nRank**: ${this.rank} (${this.statproduct})\n` +
+    `**CP**: ${this.cp} @ Level ${this.level}\n`;
+    let nameField = `**${args['pvp_pokemon'].pbName.titleCase()}**  ${this.inputATKiv}/${this.inputDEFiv}/${this.inputSTAiv}\n`; 
+    const embed = new MessageEmbed()
+    .setColor('#43B581')
+    .addField(nameField, requestInfo)
+    .setThumbnail(args['pvp_pokemon'].imageURL)
+    .setFooter(`${message.author.username} || Data retrieved from https://gostadium.club`);
+
+    let userCommand = `${message.client.commandPrefix}${command}`; //Grabs the !great, !ultra or w/e from user inputs.
+    let userPokemon = `${args['pvp_pokemon'].pbName.titleCase()}`; //Grabs the accepted pokemon name.
+    let userIVString = `${this.inputATKiv} ${this.inputDEFiv} ${this.inputSTAiv}`; //Grabs the accepted IV sets (0-15 for ATK,DEF,STA)
+    let responseCommand = `${message.author}, here are your \`${userCommand} ${userPokemon} ${userIVString}\` results:`; //Combined the above into the whole accepted command.
+    let response = await message.channel.send(responseCommand, embed);
+    response.preserve = true;
+  }
+}
+  class RankCommand extends Commando.Command {
+    constructor(client) {
+    super(client, {
+      name: 'rank',
+      group: CommandGroup.UTIL,
+      memberName: 'rank',
+      aliases: [/*'pvp-rank', 'pvp-iv', */'great', 'ultra'], //Aliases are other user commands that call this command file
+      description: 'Provides PvP data based on a pokemon species\'s IVs, including rank and CP.', //Help info
+      details: 'Use this command in #rank-checker to obtain information on the PvP ranking of a specific IV combination for a specific species of pokemon.' +

```

Comment:
We don't have `#rank-checker`, please update this to `#the-bot-lab` setting (`settings.channels['bot-lab']`).

Reply:
This command should get its own dedicated channel and be exclusive to that channel OR just allowed in w/e, similar to counters. Let it be a moderation thing, not a software thing. If you know of an easy way to store a channel ID in the DB that's set by an admin for this command, great. I could take a stab at it but it'll take me some time- I've not done coding with DB stuff in Node.js yet.

Reply:
That’s a valid argument, however, this comment still stands in principal. You can add a new key under the settings for channels for a `rank-checker` and than use that.